app/src/main/java/com/example/frogweather/ui/SettingsViewModel.kt

```diff
@@ -8,15 +8,34 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.ViewModelProvider
 import androidx.lifecycle.asLiveData
 import androidx.lifecycle.viewModelScope
+import com.example.frogweather.data.LOCATION_UPDATE_INTERVAL
+import com.example.frogweather.data.MINUTES_DIVIDER
 import com.example.frogweather.data.Settings
 import com.example.frogweather.data.SettingsDataSource
 import com.example.frogweather.data.SettingsRepository
+import com.example.frogweather.data.MyLocation
 import kotlinx.coroutines.launch
 
 class SettingsViewModel(application: Application) : AndroidViewModel(application) {
     private val settingsRepository = SettingsRepository(SettingsDataSource(application))
 
-    fun getSettings(): LiveData<Settings> = settingsRepository.getSettings().asLiveData()
+    fun getSettings(): LiveData<Settings> {
+        return settingsRepository.getSettings().asLiveData()
+    }
+
+    fun getLocation(): LiveData<MyLocation> = settingsRepository.getLocation().asLiveData()
+
+
+    fun saveLocation(location: MyLocation, context: Context) {
+        viewModelScope.launch {
+            settingsRepository.getLocation().collect { oldLocation ->
+                val minutes = (location.millis - oldLocation.millis) / MINUTES_DIVIDER
```

Comment:
What you are trying to do here?

Reply:
i check how many minutes have passed since the last location update. if it is more than 30 minutes then i will update the location again. now i moved this check in the MainActivity. checking it like this didn't prevent me from requesting the location at every startup when it was unnecessary. now in the Main activity if the last location is not older than 30 minutes i won't request a new location. is it ok?